backend/src/main/java/fr/inra/urgi/rarebasket/config/SecurityConfig.java

```diff
@@ -1,83 +1,46 @@
  * integration doesn't work in tests. This means that all MVC tests should be annotated with
  * <code>ActiveProfiles("test")</code>.
  *
- * Most of the code in this file comes from this blog post:
- * https://www.thomasvitale.com/spring-security-keycloak/
- *
  * @author JB Nizet
  */
-@Profile("!test")
-@KeycloakConfiguration
-public class SecurityConfig extends KeycloakWebSecurityConfigurerAdapter {
-
-    /**
-     * Register KeycloakAuthenticationProvider with the authentication manager.
-     * In this way, Keycloak will be responsible for providing authentication services.
-     * Spring Security has a convention to handle security roles in a format like ROLE_XX (where XXX is the actual
-     * security role name defined in Keycloak).
-     */
-    @Autowired
-    public void configureGlobal(AuthenticationManagerBuilder auth) {
-        SimpleAuthorityMapper grantedAuthorityMapper = new SimpleAuthorityMapper();
-        grantedAuthorityMapper.setPrefix("ROLE_");
-
-        KeycloakAuthenticationProvider keycloakAuthenticationProvider = keycloakAuthenticationProvider();
-        keycloakAuthenticationProvider.setGrantedAuthoritiesMapper(grantedAuthorityMapper);
-        auth.authenticationProvider(keycloakAuthenticationProvider);
-    }
-
-    /**
-     * We do not want to create any session. This should have the expected effect, but unfortunately doesn't.
-     * So a session is created anyway (but I don't know why), for nothing.
-     * If a session is necessary, replace the implementation by
-     * <code>return new RegisterSessionAuthenticationStrategy(new SessionRegistryImpl());</code>
-     */
-    @Override
-    protected SessionAuthenticationStrategy sessionAuthenticationStrategy() {
-        return new NullAuthenticatedSessionStrategy();
-    }
-
-    /**
-     * Configure the HTTP security.
-     * This is done by delegating to the standard Keycloak configuration for the technical part,
-     * and then by calling {@link #configure(HttpSecurity)} for the functional part.
-     * This public static method is there to allow the configuration used in MVC tests to use the
-     * same functional configuration by also calling this public static method.
-     */
-    @Override
-    protected void configure(HttpSecurity http) throws Exception {
-        super.configure(http);
-        configureHttp(http);
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig implements WebMvcConfigurer {
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        http.csrf(configurer -> configurer.disable())
+            .sessionManagement(customizer -> customizer.sessionCreationPolicy(SessionCreationPolicy.STATELESS))
+            .authorizeHttpRequests(
+                customizer -> customizer.requestMatchers("/api/baskets/**").permitAll()
+                                        .requestMatchers("/api/**").authenticated()
+                                        .anyRequest().permitAll()
```

Comment:
This is not exactly the previous behavior, right? Some endpoints were not authenticated (like /api/grcs) and are now?

Reply:
Indeed. But I think it was a mistake (bug) in the previous version.